torch/csrc/jit/api/module.cpp

```diff
@@ -203,18 +203,22 @@ Module Module::clone_impl(
   size_t N = type()->numAttributes();
   for (size_t i = 0; i < N; ++i) {
     IValue s = _ivalue()->getSlot(i);
-    if (type()->getAttribute(i)->is_module()) {
+    TypePtr attr_type = type()->getAttribute(i);
+    if (attr_type->is_module()) {
       const Module& orig = Module(s.toObject());
       Module cloned = orig.clone_impl(type_remap, inplace, memo);
       type_remap[orig.type()] = cloned.type();
-      r.register_module(type()->getAttributeName(i), cloned);
+      r.register_module(
```

Comment:
Concretely, we should change this from a call to `register_module` to (pseudocode):
```
r.type.addAttribute(name, interface)
r.ivalue.setattr(name, cloned_submodule)
```